mysql/Dockerfile

```diff
@@ -1,19 +1,47 @@
 FROM fedora
-MAINTAINER http://fedoraproject.org/wiki/Cloud
 
-RUN dnf -y update && dnf clean all
-RUN dnf -y install community-mysql-server community-mysql pwgen supervisor bash-completion psmisc net-tools && dnf clean all
+# MySQL image for OpenShift.
+#
+# Volumes:
+#  * /var/lib/mysql/data - Datastore for MySQL
+# Environment:
+#  * $MYSQL_USER - Database user name
+#  * $MYSQL_PASSWORD - User's password
+#  * $MYSQL_DATABASE - Name of the database to create
+#  * $MYSQL_ROOT_PASSWORD (Optional) - Password for the 'root' MySQL account
 
-ADD ./start.sh /start.sh
-ADD ./config_mysql.sh /config_mysql.sh
-ADD ./supervisord.conf /etc/supervisord.conf
+MAINTAINER http://fedoraproject.org/wiki/Cloud
 
-# RUN echo %sudo	ALL=NOPASSWD: ALL >> /etc/sudoers
+ENV MYSQL_VERSION=10.0 \
+    HOME=/var/lib/mysql
 
-RUN chmod 755 /start.sh
-RUN chmod 755 /config_mysql.sh
-RUN /config_mysql.sh
+LABEL io.k8s.description="MySQL is a multi-user, multi-threaded SQL database server" \
+      io.k8s.display-name="MySQL 5.6" \
+      io.openshift.expose-services="3306:mysql" \
+      io.openshift.tags="database,mysql,mysql56"
 
 EXPOSE 3306
 
-CMD ["/bin/bash", "/start.sh"]
+# This image must forever use UID 27 for mysql user so our volumes are
+# safe in the future. This should *never* change, the last test is there
+# to make sure of that.
+RUN dnf -y --setopt=tsflags=nodocs install gettext hostname bind-utils community-mysql-server community-mysql && \
+    dnf clean all && \
+    mkdir -p /var/lib/mysql/data && chown -R mysql.0 /var/lib/mysql && \
+    test "$(id mysql)" = "uid=27(mysql) gid=27(mysql) groups=27(mysql)"
```

Comment:
What I've seen elsewhere - including Docker docs - is creating the user in advance.  It's not hard to specify the UID, and it'll error out if there's a conflict. And it seemed to work ok in my image? The comment doesn't explain why it's done differently here, anyway.

Personally, I find the other way more obvious. Seeing the naked assertion "Fedora creates mysql with the UID 27" is surprising. Unless we want to reference an allocation document. I came across the Debian allocations when I was looking into this, hopefully there's a Fedora equivalent. As I say, my understanding is these images are nice examples people can learn and derive from, so uncommented magic numbers are not the best.

It might also help avoid having different boilerplate, dependent on whether the ID is dynamic or not...

I guess if we specified the ID, we still have to choose between referencing the allocation document, or changing it to something like 1000.

I think the nice thing about distro hard-coded ID allocations is for rescue disks. On Docker I'm not sure there's any reason not to allocate our own IDs. I guess it is a low-level interference with the Fedora package though ("layering violation").